p2p/server/msgserver.go

```diff
@@ -108,10 +109,10 @@ func (p *MessageServer) readLoop() {
 				return
 			}
 			p.workerCount.Add(1)
-			p.workerLimiter <- struct{}{}
+			//p.workerLimiter <- struct{}{}
```

Comment:
should probably un-comment but can you still explain please?

Reply:
There was a race condition that caused a deadlock in the following test: `TestFindNodeProtocol_FindNode_Concurrency` in `p2p/discovery/protocol_test.go`. I worked on this for a while and commenting the above line was a fix to it. I can keep looking into it more.

Reply:
we shouldn't merge before working this out, might be a test specific bug or some deadlock that can happen in real-time. I'll take a look once this will be the only thing left..